### PR TITLE
Add iOS simulator prompt

### PR DESCRIPTION
**Pull request by OzzieOrca**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -167,6 +179,39 @@ const setApiEnv = async (apiEnvArg: ApiEnv) => {
...
+  const devicesObj: {
+    devices: {
+      [key: string]: {
+        state: string;
+        isAvailable: boolean;
+        name: string;
+        udid: string;
+      }[];
+    };
+  } = JSON.parse(
```
Comment (soberstadt) [inline src/index.ts]: This typing of parsed JSON is awesome!

File: src/index.ts [hunk 2]
```diff
@@ -167,6 +179,39 @@ const setApiEnv = async (apiEnvArg: ApiEnv) => {
...
+
+  const { simulator }: { simulator: string } = simulatorArg
+    ? { simulator: simulatorArg }
+    : await inquirer.prompt([
+        {
+          type: 'list',
+          name: 'simulator',
+          message: 'Which iOS simulator would you like to use?'.yellow,
+          choices: simulators,
+          default: 'iPhone X',
```
Comment (soberstadt) [inline src/index.ts]: Will this be a problem on systems where iPhone X isn't defined?

File: src/index.ts [hunk 3]
```diff
@@ -167,6 +179,39 @@ const setApiEnv = async (apiEnvArg: ApiEnv) => {
...
+  } = JSON.parse(
+    (await exec('xcrun simctl list devices iPhone available -j')).stdout,
+  );
+
+  const simulators = Object.values(devicesObj.devices)
+    .flat()
+    .map(({ name }) => name);
+
+  const { simulator }: { simulator: string } = simulatorArg
+    ? { simulator: simulatorArg }
```
Comment (soberstadt) [inline src/index.ts]: should this be a guard clause at the beginning of the method to avoid having to execute the `xcrun`?

Comment (OzzieOrca) [reply src/index.ts]: Oh probably... I'll fix it

File: src/index.ts [hunk 4]
```diff
@@ -69,16 +73,22 @@ class MhQaCli extends Command {
...
 
     switch (mode) {
-      case 'ios':
-        // TODO: add API env selection
-        await prepareForBuild(args.branch, args.apiEnv);
+      case 'ios': {
+        const branch = await promptForCommonArgs(args.branch, args.apiEnv);
+        const simulator = args.simulator
+          ? args.simulator
+          : await pickIosSimulator();
```
Comment (soberstadt) [inline src/index.ts]: Why choose this syntax over `const simulator = args.simulator || await pickIosSimulator();`?

Comment (OzzieOrca) [reply src/index.ts]: That's better. I'll update it.

Probably cuz I can never remember what works when spreading. `{ ...(undefined || {}) }` works but `[ ...(undefined || []) ]` throws an error. Using a ternary lets you specify a default value when you have nothing to spread which is usually an empty array or object.

This isn't that situation but that's where the habit came from.

**Comment by soberstadt** [comment]
I'm glad you got this right and that's up to my hind standards ;)

**Pull request merged**